shell/platform/windows/windows_lifecycle_manager.h

```diff
@@ -9,12 +9,23 @@
 
 #include <cstdint>
 #include <map>
+#include <mutex>
 #include <optional>
+#include <set>
+
+#include "flutter/shell/platform/common/app_lifecycle_state.h"
 
 namespace flutter {
 
 class FlutterWindowsEngine;
 
+enum WindowStateEvent {
```

Comment:
Flutter's style seems to be `enum class` (see [this](https://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum)) and names like `kShow`, `kHide`, etc...